src/auth/index.js

```diff
@@ -0,0 +1,35 @@
+import { createCanonicalRequest, createScope, createStringToSign, createSigningKey, createSignature, createAuthorizationHeader, createXAmzContentSha256Header, createXZmzDateHeader } from './aws';
+import {
+    AWS_REGION,
+    ERROR_PARAM_REQUIRED,
+    ERROR_PARAM_TYPE_IS_NOT_OBJECT,
+    ERROR_PARAM_TYPE_IS_NOT_STRING,
+} from '../consts';
+
+export default function amendRequest(request, awsAccessKey, awsSecretKey) {
+    if (!request) throw Error(ERROR_PARAM_REQUIRED);
```

Comment:
I don't think a falsy test is good here or in any of the similar places. Better to change them all to e.g. `request === undefined` IMO...